src/sagas/WorkClaimForm.saga.ts

```diff
@@ -0,0 +1,43 @@
+import { Frost } from '@po.et/frost-client'
+import { Actions } from 'actions'
+import { delay, SagaIterator } from 'redux-saga'
+import { call, takeLatest, put, ForkEffect } from 'redux-saga/effects'
+const { toast } = require('react-toastify')
+
+import { Configuration } from 'configuration'
+
+const FrostClient = new Frost({ host: Configuration.frostApiUrl })
+
+export function WorkClaimFormSaga(): () => IterableIterator<ForkEffect> {
+  return function*(): IterableIterator<ForkEffect> {
+    yield takeLatest(Actions.WorkClaimForm.SUBMIT, WorkClaimForm)
+  }
+}
+
+export function* WorkClaimForm(action: any): SagaIterator {
```

Comment:
This should be split into much smaller watchers and handlers

Reply:
OK I think I've done it as you requested.

Reply:
much better